src/plugins/vis_type_vega/public/expressions/line_vega_spec_fn.ts

```diff
@@ -68,7 +69,22 @@ export const formatDataTable = (
   datatable.columns.forEach((column) => {
     // clean quotation marks from names in columns
     column.name = cleanString(column.name);
+    // clean ids to remove "." as that will cause vega to not process it correctly.
+    // This happens for different metric types
+    column.id = column.id.replaceAll('.', '-');
   });
+
+  // clean row keys to remove "." as that will cause vega to not process it correctly
```

Comment:
I think this is worth adding some tests for some of these scenarios & edge cases

Reply:
Added tests